mne/time_frequency/tfr.py

```diff
@@ -1512,8 +1512,13 @@ def plot_joint(self, timefreqs=None, picks=None, baseline=None,
             fmax = freq + freq_half_range
 
             data = tfr.data
-
-            pos = find_layout(tfr.info).pos if layout is None else layout.pos
+			
+            if layout is None:
+			    layout = find_layout(tfr.info)
+            
+            # only use position information for channels from layout 
+            # whose names appear as a substring in info['ch_names']
+            pos = np.array([p for p,n in zip(layout.pos,layout.names) if np.any([n in ni for ni in tfr.info['ch_names']])])
```

Comment:
something like ...

```
idx = [ii for ii, ch_name in enumerate(layout.names) if ch_name in tfr.ch_names]
pos = layout.pos[idx]
```

?

Reply:
OK, looks better, but ch_name may only be a substring of an entry in tfr.ch_names due to suffixes e.g. in CTF data.

Reply:
Ok then do the `any(` solution, but I think pos[idx] is much cleaner, don't you think?

Reply:
Yes, I agree!